src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -451,9 +451,11 @@ class ShenandoahHeap : public CollectedHeap {
 
   ShenandoahPhaseTimings*    _phase_timings;
 
-  ShenandoahControlThread*   control_thread()          { return _control_thread;    }
   ShenandoahMarkCompact*     full_gc()                 { return _full_gc;           }
 
+public:
+  ShenandoahControlThread*   control_thread() const    { return _control_thread;    }
+
```

Comment:
Maybe it is easier to `friend` the class that wants it, instead of exposing the control thread for everyone?